### PR TITLE
prohibit agg or groupby pushdown on double read

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
This solves an index bug.
### What is changed and how it works?
We prohibit agg or groupby pushdown on double read case. 

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

Related changes
 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by birdstorm** [comment]
table scan should also allow aggregate push down.

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java [hunk 1]
```diff
@@ -463,58 +464,75 @@ else if (col.getColumnInfo().isPrimaryKey() && tableInfo.isPkHandle()) {
...
-            .forEach(
-                tiExpr ->
-                    aggregationBuilder.addAggFunc(
-                        ProtoConverter.toProto(tiExpr, colOffsetInFieldMap)));
-        executorBuilder.setTp(ExecType.TypeAggregation);
-        dagRequestBuilder.addExecutors(executorBuilder.setAggregation(aggregationBuilder));
-        executorBuilder.clear();
-        addPushDownGroupBys();
-        addPushDownAggregates();
+        pushDownAggAndGroupBy(dagRequestBuilder, executorBuilder, colOffsetInFieldMap);
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java]: How about we add some comments for better understanding?

Comment (zhexuany) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java]: indded

**Pull request merged**